### PR TITLE
feat: Upgrade to TypeScript 5.5.2

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit upgrades TypeScript to 5.5.2.

https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/

### CODE REVIEW

**Comment by bartlomieju** [comment]
Ref https://github.com/denoland/TypeScript/pull/12

File: cli/tsc/dts/lib.es2021.promise.d.ts [hunk 1]
```diff
@@ -45,4 +45,10 @@ interface PromiseConstructor {
      * @returns A new Promise.
      */
     any<T>(values: Iterable<T | PromiseLike<T>>): Promise<Awaited<T>>;
+
+    /**
+     * Creates a Promise that can be resolved or rejected using provided functions.
+     * @returns An object containing `promise` promise object, `resolve` and `reject` functions.
+     */
+    withResolvers<T>(): { promise: Promise<T>; resolve: (value: T | PromiseLike<T>) => void; reject: (reason?: any) => void; };
```
Comment (petamoriken) [inline cli/tsc/dts/lib.es2021.promise.d.ts:53]: This feature should be ES2024

File: cli/tsc/dts/lib.webworker.d.ts [hunk 2]
```diff
@@ -1029,8 +1042,6 @@ interface Body {
     arrayBuffer(): Promise<ArrayBuffer>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
     blob(): Promise<Blob>;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bytes) */
-    bytes(): Promise<Uint8Array>;
```
Comment (petamoriken) [inline cli/tsc/dts/lib.webworker.d.ts:1033]: Please remain 🙏

File: cli/tsc/dts/lib.webworker.d.ts [hunk 3]
```diff
@@ -5682,11 +5694,9 @@ declare var URL: {
     prototype: URL;
     new(url: string | URL, base?: string | URL): URL;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/canParse_static) */
-    canParse(url: string | URL, base?: string | URL): boolean;
+    canParse(url: string | URL, base?: string): boolean;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/createObjectURL_static) */
     createObjectURL(obj: Blob): string;
-    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/parse_static) */
-    parse(url: string | URL, base?: string | URL): URL | null;
```
Comment (petamoriken) [inline cli/tsc/dts/lib.webworker.d.ts:5689]: ditto

**Review by petamoriken — CHANGES_REQUESTED** [review]

File: cli/tsc/dts/lib.es2021.promise.d.ts (hunk 1 above)
Comment (petamoriken) [reply cli/tsc/dts/lib.es2021.promise.d.ts:53]: As of TypeScript 5.5.2, it seems to be defined in [lib.esnext.promise.d.ts](https://github.com/microsoft/TypeScript/blob/v5.5.2/lib/lib.esnext.promise.d.ts). 

I work in progress to move this to lib.es2024.promise.d.ts.
https://github.com/microsoft/TypeScript/pull/58573

**Review by bartlomieju — COMMENTED** [review]
Thanks for the review @petamoriken, I reverted these changes.

File: cli/tsc/dts/lib.dom.d.ts [hunk 4]
```diff
@@ -22739,7 +22909,7 @@ declare var URL: {
     prototype: URL;
     new(url: string | URL, base?: string | URL): URL;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/canParse_static) */
-    canParse(url: string | URL, base?: string | URL): boolean;
+    canParse(url: string | URL, base?: string): boolean;
```
Comment (petamoriken) [inline cli/tsc/dts/lib.dom.d.ts:22912]: please fix 🙏

File: cli/tsc/dts/lib.webworker.d.ts [hunk 5]
```diff
@@ -5684,7 +5698,7 @@ declare var URL: {
     prototype: URL;
     new(url: string | URL, base?: string | URL): URL;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URL/canParse_static) */
-    canParse(url: string | URL, base?: string | URL): boolean;
+    canParse(url: string | URL, base?: string): boolean;
```
Comment (petamoriken) [inline cli/tsc/dts/lib.webworker.d.ts:5701]: ditto

**Review by petamoriken — CHANGES_REQUESTED** [review]

**Review by dsherret — APPROVED** [review]
LGTM!

**Comment by petamoriken** [comment]
I updated upstream types 😄
https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1755

**Comment by bartlomieju** [comment]
Very cool, thanks!

**Pull request merged**